File: iris-core/tests/fixtures/symbols/sample.cpp
/**
 * Maximum retry count.
 */
const int MAX_RETRIES = 3;

/**
 * Application configuration.
 */
class AppConfig {
public:
    std::string name;
    bool debug;

    AppConfig(std::string name, bool debug)
        : name(std::move(name)), debug(debug) {}

    bool isDebug() const {
        return debug;
    }
};

/**
 * Color options.
 */
enum Color {
    RED,
    GREEN,
    BLUE
};

/**
 * Greet a user by name.
 */
void greet(const std::string &name) {
    std::cout << "Hello, " << name << "!" << std::endl;
}

File: iris-core/tests/fixtures/symbols/sample.rs
/// Configuration for the application.
pub struct AppConfig {
    pub name: String,
    pub debug: bool,
}

/// Color options.
pub enum Color {
    Red,
    Green,
    Blue,
}

/// A trait for serialization.
pub trait Serialize {
    fn serialize(&self) -> String;
}

/// Maximum retry count.
pub const MAX_RETRIES: u32 = 3;

/// Greet a user by name.
pub fn greet(name: &str) -> String {
    format!("Hello, {name}!")
}

impl AppConfig {
    /// Create a new config with defaults.
    pub fn new() -> Self {
        Self {
            name: String::new(),
            debug: false,
        }
    }
}
